src/main/duke/Duke.java

```diff
@@ -0,0 +1,391 @@
+package main.duke;
+
+import main.duke.exception.DukeException;
+import main.duke.task.Deadline;
+import main.duke.task.Event;
+import main.duke.task.Task;
+
+import java.util.ArrayList;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
+import java.util.regex.Pattern;
+
+public class Duke {
+
+    /* Variables for Duke's use */
+    private static ArrayList<Task> taskList;
+    private static int listIndex;
+    private static Scanner scanner;
+
+    /* Constants */
+    private static final String INDENT = "\n    ";
+    private static final String H_LINE = INDENT +
+            "------------------------------------------------";
+    private static final int TASK_LIMIT = 100;
+
+
+    public static void main(String[] args) {
+
+        //Initialize necessary variables
+        init();
```

Comment:
Perhaps consider renaming this function to something more intuitive so the comment won't have to explain what it does.